vendor/github.com/manicminer/hamilton/msgraph/groups.go

```diff
@@ -0,0 +1,451 @@
+			ValidStatusFunc:  checkMemberGone,
+			Uri: Uri{
+				Entity:      fmt.Sprintf("/groups/%s/members/%s/$ref", id, memberId),
+				HasTenantId: true,
+			},
+		})
+		if err != nil {
+			return status, fmt.Errorf("GroupsClient.BaseClient.Delete(): %v", err)
+		}
+	}
+	return status, nil
+}
+
+// ListOwners retrieves the owners of the specified Group.
+// id is the object ID of the group.
+func (c *GroupsClient) ListOwners(ctx context.Context, id string) (*[]string, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/groups/%s/owners", id),
+			Params:      url.Values{"$select": []string{"id"}},
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("GroupsClient.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("ioutil.ReadAll(): %v", err)
+	}
+	var data struct {
+		Owners []struct {
+			Type string `json:"@odata.type"`
+			Id   string `json:"id"`
+		} `json:"value"`
+	}
+	if err := json.Unmarshal(respBody, &data); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	ret := make([]string, len(data.Owners))
+	for i, v := range data.Owners {
+		ret[i] = v.Id
+	}
+	return &ret, status, nil
+}
+
+// GetOwner retrieves a single owner for the specified Group.
+// groupId is the object ID of the group.
+// ownerId is the object ID of the owning object.
+func (c *GroupsClient) GetOwner(ctx context.Context, groupId, ownerId string) (*string, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/groups/%s/owners/%s/$ref", groupId, ownerId),
+			Params:      url.Values{"$select": []string{"id,url"}},
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("GroupsClient.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("ioutil.ReadAll(): %v", err)
+	}
+	var data struct {
+		Context string `json:"@odata.context"`
+		Type    string `json:"@odata.type"`
+		Id      string `json:"id"`
+		Url     string `json:"url"`
+	}
+	if err := json.Unmarshal(respBody, &data); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	return &data.Id, status, nil
+}
+
+// AddOwners adds a new owner to a Group.
+// First populate the Owners field of the Group using the AppendOwner method of the model, then call this method.
+func (c *GroupsClient) AddOwners(ctx context.Context, group *Group) (int, error) {
+	var status int
+	for _, owner := range *group.Owners {
+		// don't fail if an owner already exists
+		checkOwnerAlreadyExists := func(resp *http.Response, o *odata.OData) bool {
```

Comment:
wouldn't this indicate a bug?

Reply:
This mostly helps to work around API inconsistency where newly created groups (and other objects) sometimes inherit an owner but sometimes don't